src/plugins/intel_gpu/src/runtime/kernels_cache.cpp

```diff
@@ -579,13 +586,23 @@ kernels_cache::compiled_kernels kernels_cache::compile(const kernel_impl_params&
     std::vector<batch_program> batches;
     get_program_source(t_kernels_code, &batches);
 
-    compiled_kernels output_kernels;
+    compiled_kernels _compiled_kernels;
     // Build batches
     for (size_t idx = 0; idx < batches.size(); ++idx) {
-        build_batch(_build_engine, batches[idx], output_kernels);
+        build_batch(_build_engine, batches[idx], _compiled_kernels);
+    }
+
+    OPENVINO_ASSERT(_compiled_kernels.size() == 1, "Only the kernels of the single primitive should be compiled.");
```

Comment:
Could you rename the function to be more specific, like compile_single_impl ?

Reply:
Fixed it.